src/v2/view-builder/views/DeviceChallengePollView.js

```diff
@@ -0,0 +1,117 @@
+/* global Promise */
+import { _, $ } from 'okta';
+import BaseView from '../internals/BaseView';
+import BaseForm from '../internals/BaseForm';
+
+const request = (url, method = 'GET', body = null, timeout = 1000) => {
+  return $.ajax({
+    url,
+    method,
+    timeout,
+    contentType: 'application/json',
+    data: body,
+  });
+};
+
+const Body = BaseForm.extend({
+  noButtonBar: true,
+
+  className: 'ion-form device-challenge-poll',
+
+  initialize () {
+    BaseForm.prototype.initialize.apply(this, arguments);
+    this.deviceChallengePollRemediation = this.options.appState.get('remediation').find(
+      rem => rem.name === 'device-challenge-poll'
+    );
+    this.doChallenge();
+    this.startPolling();
+  },
+
+  remove () {
+    BaseForm.prototype.remove.apply(this, arguments);
+    this.stopPolling();
+  },
+
+  doChallenge () {
+    const deviceChallengeHintsKey = this.deviceChallengePollRemediation.relatesTo;
+    const deviceChallenge = deviceChallengeHintsKey &&
```

Comment:
nit:
doesn't have to check exists of `deviceChallengeHintsKey` cause `x[undefined] === undefined`